.github/workflows/build-macos.yml

```diff
@@ -59,59 +59,26 @@ jobs:
         id: paths
         run : |
           echo "build_dir=${{ github.workspace }}/build" >> $GITHUB_OUTPUT
-          echo "ext_deps_dir=${{ github.workspace }}/external_dependencies" >> $GITHUB_OUTPUT
           echo "install_dir=${{ github.workspace }}/install" >> $GITHUB_OUTPUT
 
-      # Step: Install system-provided dependencies
-      # macOS
-      - if: runner.os == 'macOS'
-        name: Install system-provided dependencies
-        run: | # brew update
-          brew install boost
-          brew install doxygen
-      # Linux
-      - if: runner.os == 'Linux'
-        name: Install system-provided dependencies
-        run: | # sudo apt-get update && sudo apt upgrade
-          sudo apt-get install libboost-all-dev
-          sudo apt-get install doxygen
-      
-      # Step: Restore cached user-provided dependencies
-      - name: Restore cached user-provided dependencies
-        uses: actions/cache/restore@v3
-        id: restore-cached-external-dependencies
-        with:
-          key: ${{ runner.os }}-cache-key
-          restore-keys: ${{ runner.os }}-cache-key
-          path: ${{ steps.paths.outputs.ext_deps_dir }}/netcdf-c/install/netcdf-c
-
-      # Step: Build and install user-provided dependencies, executes only if no cache restored
-      - name: Build and install user-provided dependencies
-        if: steps.restore-cached-external-dependencies.outputs.cache-hit != 'true'
-        # NetCDF Dependencies m4, curl, and openssl are provided by the build machine
-        run: >
-          pwsh ${{ github.workspace }}/scripts/install_netcdf_static.ps1
-          -WorkDir ${{ steps.paths.outputs.ext_deps_dir }}/netcdf-c/work
-          -InstallDir ${{ steps.paths.outputs.ext_deps_dir }}/netcdf-c/install
-          -BuildType '${{ matrix.build_type }}'
-          -ParallelJobs 10
+      - name: Install system-provided dependencies
+        run: |
+          if [ "${{ runner.os }}" == "macOS" ]; then
```

Comment:
while we're at it, let's make the version coherent with the TC linux build:

- boost@1.81.0
- netcdf@4.9.1

Reply:
1.81 doesn't exist on brew it seems, 1.83 does.

Reply:
But it turns out we can't set versions on brew :/

Reply:
It seems that the minor version is not needed for boost.  Please try `brew install boost@1.81`